src/index.js

```diff
@@ -1,65 +1,172 @@
+
 /**
- * App ID for the skill
+ * This sample demonstrates a simple skill built with the Amazon Alexa Skills Kit.
+ * The Intent Schema, Custom Slots, and Sample Utterances for this skill, as well as
+ * testing instructions are located at http://amzn.to/1LzFrj6
+ *
+ * For additional samples, visit the Alexa Skills Kit Getting Started guide at
+ * http://amzn.to/1LGWsLG
  */
 
-var APP_ID = "amzn1.echo-sdk-ams.app.23ed8a00-90ef-4ba5-b418-a1d4d8ec5e32";
-
-var AlexaSkill = require('./AlexaSkill');
-
-
+// Route the incoming request based on type (LaunchRequest, IntentRequest,
+// etc.) The JSON body of the request is provided in the event parameter.
 
-var ReverseGeek = function () {
-    AlexaSkill.call(this, APP_ID);
+exports.handler = function (event, context) {
+    try {
+        console.log("event.session.application.applicationId=" + event.session.application.applicationId);
+
+
+        if (event.session.application.applicationId !== "amzn1.echo-sdk-ams.app.23ed8a00-90ef-4ba5-b418-a1d4d8ec5e32") {
+             context.fail("Invalid Application ID");
+        }
+
+        if (event.session.new) {
```

Comment:
yes

On Fri, Jun 10, 2016 at 1:31 PM Javier Forero <notifications@github.com>
wrote:

> In src/index.js
> <https://github.com/javierforero/reverse_geek/pull/1#discussion_r66651159>
> :
>
> >
> > -var ReverseGeek = function () {
> > -    AlexaSkill.call(this, APP_ID);
> > +exports.handler = function (event, context) {
> > +    try {
> > +        console.log("event.session.application.applicationId=" + event.session.application.applicationId);
> > +
> > +
> > +        if (event.session.application.applicationId !== "amzn1.echo-sdk-ams.app.23ed8a00-90ef-4ba5-b418-a1d4d8ec5e32") {
> > +             context.fail("Invalid Application ID");
> > +        }
> > +
> > +        if (event.session.new) {
>
> humm. I am not really sure what this is doing. Is this saying if this is a
> new session call onSessionStarted() and pass it an object containing
> requestId and the event session? The onSessionStarted() is just printing
> the requestId and session.id out onto the console?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/javierforero/reverse_geek/pull/1/files/d9d8d7dbf411b8fbc9a081a7ed82fc05b47c2fb6#r66651159>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEuwTkoS_ww-GidRqXnAdfWSgJ1ZArVqks5qKZ9ngaJpZM4Iw72E>
> .
>